### PR TITLE
fix(FEC-11606): merge active cue-points from all metadata TextTracks

### PR DESCRIPTION
**Pull request by semarche-kaltura**
### Description of the Changes

Merge active cue-points from all metadata TextTracks for TIMED_METADATA payload

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/engines/html5/html5.js [hunk 1]
```diff
@@ -1165,7 +1165,13 @@ export default class Html5 extends FakeEventTarget implements IEngine {
...
       track.mode = PKTextTrack.MODE.HIDDEN;
       this._eventManager.listen(track, 'cuechange', () => {
-        this.dispatchEvent(new FakeEvent(CustomEventType.TIMED_METADATA, {cues: Array.from(track.activeCues), label: track.label}));
+        let activeCues = [];
+        Array.from(this._el.textTracks).forEach((track: TextTrack) => {
+          if (PKTextTrack.isMetaDataTrack(track)) {
+            activeCues = activeCues.concat(Array.from(track.activeCues));
+          }
+        });
+        this.dispatchEvent(new FakeEvent(CustomEventType.TIMED_METADATA, {cues: activeCues}));
```
[the remark this answers is not in the document]
Comment (OrenMe) [reply src/engines/html5/html5.js:1174]: @yairans @semarche-kaltura any update?

**Comment by semarche-kaltura** [comment]
@OrenMe, @yairans PR updated

File: src/engines/html5/html5.js [hunk 2]
```diff
@@ -1167,7 +1167,13 @@ export default class Html5 extends FakeEventTarget implements IEngine {
     const listenToCueChange = track => {
       track.mode = PKTextTrack.MODE.HIDDEN;
       this._eventManager.listen(track, 'cuechange', () => {
-        this.dispatchEvent(new FakeEvent(CustomEventType.TIMED_METADATA, {cues: Array.from(track.activeCues), label: track.label}));
+        let activeCues = [];
+        Array.from(this._el.textTracks).forEach((track: TextTrack) => {
+          if (PKTextTrack.isMetaDataTrack(track)) {
+            activeCues = activeCues.concat(getActiveCues(track.activeCues));
```
Comment (yairans) [inline src/engines/html5/html5.js:1173]: as already discussed the cues should be sorted

**Review by yairans — CHANGES_REQUESTED** [review]

**Review by yairans — APPROVED** [review]

File: src/engines/html5/html5.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (semarche-kaltura) [reply src/engines/html5/html5.js:1180]: @OrenMe PR updated

File: src/engines/html5/html5.js [hunk 3]
```diff
@@ -1167,7 +1168,16 @@ export default class Html5 extends FakeEventTarget implements IEngine {
...
+          }
+        });
+        activeCues = activeCues.sort((a: Cue, b: Cue) => {
+          return a.startTime - b.startTime;
+        });
+        this.dispatchEvent(new FakeEvent(CustomEventType.TIMED_METADATA, {cues: activeCues}));
       });
     };
     const metadataTrack = Array.from(this._el.textTracks).find((track: TextTrack) => PKTextTrack.isMetaDataTrack(track));
```
Comment (yairans) [inline src/engines/html5/html5.js:1183]: need to change this `find` to `filter` as we now have multiple metadata tracks.
I don't know how it's working with id3 and cuepoints. maybe it get to the 'addtrack' event in the 'else' block below

**Review by yairans — CHANGES_REQUESTED** [review]

**Comment by yairans** [comment]
implemented as part of https://github.com/kaltura/playkit-js/pull/623

**Pull request closed without merging**